From 326b82bbc418519bd07ad0d69a96e53cde55a5ae Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Tue, 25 Oct 2016 23:26:12 +0200
Subject: [PATCH 1/2] Add rules required for TARGET_HAS_LEGACY_CAMERA_HAL1

Author: Adrian DC <radian.dc@gmail.com>
Date:   Tue Oct 17 23:39:27 2017 +0200

    mediaserver: Allow finding the hal_camera hardware service

     * denied { find } for
        interface=android.hardware.camera.provider::ICameraProvider
        scontext=u:r:mediaserver:s0
        tcontext=u:object_r:hal_camera_hwservice:s0 tclass=hwservice_manager

    Change-Id: I5f61a5879f74ea4f1e29246784af512fea8f8127

Change-Id: Ica4926fa3acf88b95c687f25eec4dbc68ea41fbb
---
 definitions.mk        | 1 +
 public/mediaserver.te | 7 +++++++
 2 files changed, 8 insertions(+)

diff --git a/definitions.mk b/definitions.mk
index 1af4c14..fe48996 100644
--- a/definitions.mk
+++ b/definitions.mk
@@ -9,6 +9,7 @@ $(hide) m4 $(PRIVATE_ADDITIONAL_M4DEFS) \
 	-D target_arch=$(PRIVATE_TGT_ARCH) \
 	-D target_with_asan=$(PRIVATE_TGT_WITH_ASAN) \
 	-D target_full_treble=$(PRIVATE_FULL_TREBLE) \
+	-D target_has_legacy_camera_hal1=$(TARGET_HAS_LEGACY_CAMERA_HAL1) \
 	-D target_needs_platform_text_relocations=$(TARGET_NEEDS_PLATFORM_TEXT_RELOCATIONS) \
 	$(PRIVATE_TGT_RECOVERY) \
 	-s $^ > $@
diff --git a/public/mediaserver.te b/public/mediaserver.te
index 6efaf0f..0706aec 100644
--- a/public/mediaserver.te
+++ b/public/mediaserver.te
@@ -98,6 +98,13 @@ allow mediaserver mediadrmserver_service:service_manager find;
 # For interfacing with OMX HAL
 allow mediaserver hidl_token_hwservice:hwservice_manager find;
 
+ifelse(target_has_legacy_camera_hal1, `true',
+  allow mediaserver cameraproxy_service:service_manager find;
+  allow mediaserver cameraserver_service:service_manager add;
+  allow mediaserver hal_camera_hwservice:hwservice_manager find;
+,
+)
+
 # /oem access
 allow mediaserver oemfs:dir search;
 allow mediaserver oemfs:file r_file_perms;
-- 
2.7.4

